client/core/trade.go

```diff
@@ -1612,90 +1630,107 @@ func (c *Core) redeemMatchGroup(t *trackedTrade, matches []*matchTracker, errs *
 		)
 	}
 
-	// Send the redemption information to the DEX.
+	// Save redemption details and send the redeem message to the DEX.
+	// Saving the redemption details now makes it possible to resend the
+	// `redeem` reqeust at a later time if sending it now fails.
 	for i, match := range matches {
-		err := c.finalizeRedeemAction(t, match, coinIDs[i])
-		if err != nil {
-			errs.addErr(err)
+		proof := &match.MetaData.Proof
+		coinID := []byte(coinIDs[i])
+		if match.Match.Side == order.Taker {
+			match.SetStatus(order.MatchComplete) // could this cause the match to be retired before the `redeem` request succeeds?
```

Comment:
> could this cause the match to be retired before the `redeem` request succeeds?

I think so. While it's critical to set the coin here, I think the status update could wait until after the `redeem` message.

Or maybe you just need to check the `RedeemSig` before retiring an order, in which case you would need to carefully revoke the match for most errors from the `redeem` request.

Reply:
Actually, we're already checking the `RedeemSig` in `matchIsActive`.

Reply:
`DEXOrdersWithActiveMatches` does not check the `redeemSig` if the match status is `MatchComplete` and a dexc restart will cause this match to not be loaded. I'm re-considering updating the match status only after the `redeem` request succeeds.

Reply:
How bout fixing `DEXOrdersWithActiveMatches` to work the same way as `matchIsActive`?

Reply:
I've always thought both methods should behave similarly, but will wait to see what @buck54321 thinks.

Reply:
Reflecting matrix discussion, I'd prefer that we not mark anything as `MatchComplete` until all step are done. We can infer the sub-step from the coin IDs and sigs in the match metadata.

Reply:
I'm on board with handling `MatchComplete` as a special status for now, whereby it is only updated after all steps, with the longer-term goal of creating a `TakerRedeemed` that is consistent with the other statues that reflect the transaction broadcast state.  Both changes have potential to break things in lots of ways, but just changing `MatchComplete` shouldn't be too bad.  It will of course require attention to the match status resolvers however.

Reply:
Resolving this in #957.